### PR TITLE
maxx: 1.1.0 -> 2.0.1

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/maxx/default.nix [hunk 1]
```diff
@@ -81,7 +73,7 @@ in stdenv.mkDerivation {
     homepage = http://www.maxxinteractive.com;
```
Comment (markuskowa) [inline pkgs/desktops/maxx/default.nix]:
```suggestion
    homepage = http://maxxdesktop.arcadedaydream.com;
```

Comment (gnidorah) [reply pkgs/desktops/maxx/default.nix]: @markuskowa thanks! Maxx.arcadedaydream.com contains only outdated pages. I've pointed to their facebook instead, which is their current news source

**Comment by gnidorah** [comment]
I've dropped libredirect hacks in favor of creation of symlink from package in nix store to /opt/MaXX in maxx module like it was somewhere in the past. The reason is that icons of minimized programs on desktop are broken otherwise. I wasn't able to figure issue with strace and got no help in that regard from maxx author

**Comment by markuskowa** [comment]
It seems to work so far. Here are a couple of things:

* `hinv` requires `dmidecode`, `pciutils`, and `usbutils`
* The symbols in the iconbox look broken  
 (might be my Virtualbox graphics)
* Most entries in the tools chest menu do not work. For example "System Info", "Desktop Manager", "Search for Files"

**Comment by gnidorah** [comment]
@markuskowa Thank you!
1. fixed
2. unfortunately don't know how to fix. I'll ask maxx author. As a workaround if we will not find a solution we could fallback to the previous release behaviour `*useIconBox:      False` in Xdefaults
3. these functions are not finished, and so these items point to author specific environment tools
```
$ grep -e "System Info" -e "Desktop Manager" -e "Search For Files" result/opt/MaXX/etc/system.chestrc
        "System Info"           f.checkexec.sh.le "gnome-control-center info"
        "Desktop Manager"       f.checkexec.sh.le "gnome-control-center"
        "Search For Files"      f.checkexec.sh.le "gnome-search-tool"
```
These could be overriden in `~/.chestrc`

File: nixos/modules/services/x11/desktop-managers/maxx.nix [hunk 2]
```diff
@@ -13,6 +13,12 @@ in {
   config = mkIf (xcfg.enable && cfg.enable) {
     environment.systemPackages = [ pkgs.maxx ];
 
+    # there is hardcoded path in binaries
+    system.activationScripts.setup-maxx = ''
+      mkdir -p /opt
+      ln -sfn ${pkgs.maxx}/opt/MaXX /opt
```
Comment (markuskowa) [inline nixos/modules/services/x11/desktop-managers/maxx.nix]: That is not very pretty. Is there no other way to solve this?

**Comment by markuskowa** [comment]
> 2. unfortunately don't know how to fix. I'll ask maxx author. As a workaround if we will not find a solution we could fallback to the previous release behaviour *useIconBox: False in Xdefaults. They will be shown on the desktop without issues like on this screen https://www.facebook.com/maxxdesktop/photos/rpp.1083070501713918/1557043887649908/?type=3&theater

Setting the old behavior is probably the more clean version for now.

File: nixos/modules/services/x11/desktop-managers/maxx.nix (hunk 2 above)
Comment (gnidorah) [reply nixos/modules/services/x11/desktop-managers/maxx.nix]: @markuskowa Using `libredirect`, but that way minimized icons are totally broken. Also MaXX author strongly suggests using /opt/MaXX location.

Comment (markuskowa) [reply nixos/modules/services/x11/desktop-managers/maxx.nix]: It's unfortunate that the package is closed source.

**Comment by gnidorah** [comment]
@markuskowa Thank you!

**Pull request merged**